src/test/java/teammates/test/cases/webapi/UpdateFeedbackResponseActionTest.java

```diff
@@ -155,10 +201,263 @@ public void testExecute_instructorFeedbackSubmissionMcqGenerateOptionsForTeams_s
+    }
+
+    @Test
+    protected void testAccessControl_wrongGiver_inaccessible() {
+
+        ______TS("wrong giver type");
+
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        String[] wrongGiverTypeParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+        };
+
+        verifyCannotAccess(wrongGiverTypeParams);
+    }
+
+    @Test
+    protected void testAccessControl_previewMode_inaccessible() {
+
+        ______TS("preview mode, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+
+        String[] previewParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+                Const.ParamsNames.PREVIEWAS, instructor1OfCourse1.email,
+        };
+
+        verifyCannotAccess(previewParams);
+
+    }
+
+    @Test
+    protected void testAccessControl_responseSessionNotOpen_inaccessible() {
+
+        ______TS("response in session not open, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+
+        assertFalse(closedSession.isOpened());
+
+        String[] sessionNotOpenParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, responseInClosedSession.getId(),
+        };
+
+        verifyCannotAccess(sessionNotOpenParams);
+    }
+
+    @Test
+    protected void testAccessControl_containsQuestionNotForInstructor_inaccessible() {
+
+        ______TS("Response contains question not intended shown to instructor, "
+                + "moderated instructor should not be accessible");
+
+        loginAsInstructor(instructor1OfCourse1.googleId);
+
+        assertFalse(logic.getFeedbackQuestion(typicalResponse2.getFeedbackQuestionId())
+                .getShowResponsesTo().contains(FeedbackParticipantType.INSTRUCTORS));
+
+        String[] invalidModeratedInstructorSubmissionParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse2.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_MODERATED_PERSON, instructor1OfCourse1.getEmail(),
+        };
+
+        verifyCannotAccess(invalidModeratedInstructorSubmissionParams);
+    }
+
+    @Test
+    protected void testAccessControl_noFeedbackResponse_inaccessible() {
+
+        ______TS("non-existent feedback response");
+
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        String[] nonExistParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, "randomNonExistId",
```

Comment:
Similar to the other tests, assert that a feedback response with ID "randomNonExistId" doesn't exist.

Reply:
Just checking my understanding on this, but isn't this already being done by this test case in the [`assertThrows()`](https://github.com/TEAMMATES/teammates/pull/9930/files#diff-c2c40d39719d41784a3862fc127e670bR301) method call?

Reply:
Not exactly, the assertThrows method call verifies that the method says that the feedback response doesn't exist. However, we want to verify that this is indeed correct (i.e. what if there's actually a feedback response ID with "randomNonExistId"? This is similar to your `testAccessControl_studentAccessOtherTeam_inaccessible` method, where before the check you verify that (1) the question is indeed for responding as a team and (2) the team giving the response is not the team of the logged in student.